File: topics/control-flow/exercises/ColorPick.pas
procedure DrawColoredRects();
begin
  repeat
    ProcessEvents();
    ClearScreen();
    
    if KeyDown(vk_R) then FillRectangle(ColorRed, 10, 10, 780, 580);
    if KeyDown(vk_G) then FillRectangle(ColorGreen, 20, 20, 760, 560);
    if KeyDown(vk_B) then FillRectangle(ColorBlue, 30, 30, 740, 540);

    RefreshScreen();
  until KeyTyped(vk_ESCAPE);
end;

File: topics/control-flow/exercises/color_pick.c
void draw_colored_rects()
{
    do
    {
        process_events();
        clear_screen();
        
        if (key_down(VK_R)) fill_rectangle(ColorRed, 10, 10, 780, 580);
        if (key_down(VK_G)) fill_rectangle(ColorGreen, 20, 20, 760, 560);
        if (key_down(VK_B)) fill_rectangle(ColorBlue, 30, 30, 740, 540);
        
        refresh_screen();
    } while ( ! key_typed(VK_ESCAPE) );
}